app/qmlV2/component/MMAbstractEditor.qml

```diff
@@ -0,0 +1,181 @@
+  property string warningMsg
+  property string errorMsg
+  property bool hasFocus: false
+  property color bgColor: StyleV2.whiteColor
+  property bool hasCheckbox: false
+  property bool checkboxChecked: false
+
+  readonly property real spacing: 15 * __dp
+
+  width: parent.width
+  height: mainColumn.height
+
+  Column {
+    id: mainColumn
+
+    spacing: 6 * __dp
+    anchors.left: parent.left
+    anchors.right: parent.right
+
+    Row {
+      id: titleRow
+
+      spacing: 4 * __dp
+      width: parent.width
+      visible: titleItem.text.length > 0
+
+      MMCheckBox {
+        id: checkbox
+
+        width: 16 * __dp
+        height: 16 * __dp
+        visible: root.hasCheckbox
+        checked: root.checkboxChecked
+      }
+      Text {
+        id: titleItem
+
+        width: parent.width - checkbox.width - titleRow.spacing
+        font: StyleV2.p6
+        wrapMode: Text.WordWrap
+      }
+    }
+
+    Item {
+      height: 50 * __dp
+      anchors.left: parent.left
+      anchors.right: parent.right
+
+      Rectangle {
+        id: background
+
+        width: parent.width
+        height: parent.height
+        border.color: root.hasFocus ? ( errorMsg.length > 0 ? StyleV2.negativeColor : warningMsg.length > 0 ? StyleV2.warningColor : StyleV2.forestColor ) : StyleV2.transparentColor
+        border.width: 2 * __dp
+        color: root.bgColor
+        radius: 12 * __dp
+      }
+
+      Row {
+        height: parent.height
+        anchors.left: parent.left
+        anchors.right: parent.right
+        anchors.leftMargin: root.spacing
+        anchors.rightMargin: root.spacing
+
+        Item {
+          id: leftActionContainer
+
+          property bool actionAllowed: leftActionContainer.children.length > 1 && (leftActionContainer.children[1]?.visible ?? false)
+
+          height: parent.height
+          width: actionAllowed ? height/2 : 0
+
+          Rectangle {
+            color: StyleV2.transparentColor
+            anchors.centerIn: parent
+            width: leftActionContainer.actionAllowed ? parent.width + root.spacing/2 : 0
+            height: parent.height
+
+            MouseArea {
+              anchors.fill: parent
+              onReleased: root.leftActionClicked()
+            }
+          }
+        }
+
+        Item {
+          id: contentContainer
+
+          height: parent.height
+          width: parent.width - (leftActionContainer.actionAllowed ? leftActionContainer.width : 0) - (rightActionContainer.actionAllowed ? rightActionContainer.width : 0) //- 2 * root.spacing - ( leftActionContainer.actionAllowed ? root.spacing : 0 )
+
+          MouseArea {
+            anchors.fill: parent
+
+            onClicked: {
+              root.contentClicked()
+            }
+          }
+        }
+
+        Item {
+          id: rightActionContainer
+
+          property bool actionAllowed: rightActionContainer.children.length > 1 && (rightActionContainer.children[1]?.visible ?? false)
```

Comment:
What type of input changes the visibility state? If that is the searchbar, I would hide the icon in the searchbar component, but keep the reserved space here

```suggestion
          property bool actionAllowed: rightActionContainer.children.length
```